tests/failpoints/test_io_error.rs

```diff
@@ -138,88 +137,113 @@ fn test_file_rotate_error() {
     let fs = Arc::new(ObfuscatedFileSystem::default());
     let entry = vec![b'x'; 1024];
 
-    let engine = Engine::open_with_file_system(cfg.clone(), fs.clone()).unwrap();
-    engine
+    let mut engine = Some(Engine::open_with_file_system(cfg.clone(), fs.clone()).unwrap());
+    let mut engine_ref = engine.as_ref().unwrap();
```

Comment:
No need, you can re-assign a variable after it's moved, e.g. `drop(engine); engine = Engine::new();`